src/XCCDF_POLICY/xccdf_policy_remediate.c

```diff
@@ -493,39 +499,58 @@ int xccdf_policy_rule_result_remediate(struct xccdf_policy *policy, struct xccdf
 	if (xccdf_rule_result_get_result(rr) != XCCDF_RESULT_FAIL)
 		return 0;
 
+	// if a miscellaneous error happens (fix unsuitable or if we want to skip it for any reason
+	// we set misc_error to one, and the fix will be reported as error (and not skipped without log like before)
+	int misc_error=0;
+
 	if (fix == NULL) {
 		fix = _find_suitable_fix(policy, rr);
-		if (fix == NULL)
-			// We may want to append xccdf:message about missing fix.
-			return 0;
+		if (fix == NULL) {
+			// We want to append xccdf:message about missing fix.
+			_rule_add_info_message(rr, "No suitable fix found.");
+			xccdf_rule_result_set_result(rr, XCCDF_RESULT_ERROR);
+			misc_error=1;
+		}
 	}
 
 	struct xccdf_check *check = NULL;
 	struct xccdf_check_iterator *check_it = xccdf_rule_result_get_checks(rr);
 	while (xccdf_check_iterator_has_more(check_it))
 		check = xccdf_check_iterator_next(check_it);
 	xccdf_check_iterator_free(check_it);
-	if (check != NULL && xccdf_check_get_multicheck(check))
+	
+	// this following condition seems to be unwanted, no standard specifies that
+	// a fix shouldn't be ran for a multicheck.
+	
+	/*if (check != NULL && xccdf_check_get_multicheck(check)) {
 		// Do not try to apply fix for multi-check.
-		return 0;
-
-	/* Initialize the fix. */
-	struct xccdf_fix *cfix = xccdf_fix_clone(fix);
-	int res = xccdf_policy_resolve_fix_substitution(policy, cfix, rr, test_result);
-	xccdf_rule_result_add_fix(rr, cfix);
-	if (res != 0) {
-		_rule_add_info_message(rr, "Fix execution was aborted: Text substitution failed.");
-		return res;
-	}
-
-	/* Execute the fix. */
-	res = _xccdf_fix_execute(rr, cfix);
-	if (res != 0) {
-		_rule_add_info_message(rr, "Fix was not executed. Execution was aborted.");
-		return res;
+		_rule_add_info_message(rr, "cannot apply fix for multicheck.");
+		xccdf_rule_result_set_result(rr, XCCDF_RESULT_ERROR);
+		misc_error=1;
+	}*/
```

Comment:
No need to leave the commented code here, we have git history. Please remove this including the comment.